devsearch/preprocessing/extract.py

```diff
@@ -0,0 +1,116 @@
+import difflib
+import json
+from typing import Dict
+
+from dulwich import porcelain
+from dulwich.objects import Commit, ShaFile
+from dulwich.repo import Repo
+from dulwich.walk import WalkEntry
+from tqdm import tqdm
+
+SUPPORTED_EXTENSIONS = {".py"}
+
+
+def handle_blob(old_blob, new_blob) -> Dict:
+    """
+    A method that gives blob differences
+    Args:
+        old_blob: one blob
+        new_blob: another blob
+
+    Returns: {"added": rows_added, "deleted": rows_deleted}
+
+    """
+    differences = difflib.unified_diff(old_blob.data.decode().splitlines(), new_blob.data.decode().splitlines())
+
+    rows_added = 0
+    rows_deleted = 0
+    for el in list(differences):
+        if el.startswith("+") and not el.startswith("++"):
+            rows_added += 1
+        elif el.startswith("-") and not el.startswith("--"):
+            rows_deleted += 1
+
+    return {"added": rows_added, "deleted": rows_deleted}
+
+
+def handle_entry(repo: Repo, entry: WalkEntry):
```

Comment:
let's specify return type here
```suggestion
def handle_entry(repo: Repo, entry: WalkEntry) -> List[Dict[String, Any]]:
```